docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile

```diff
@@ -1,7 +1,28 @@
 # Build toolchains on ubuntu22.04, dev-env image can be built based on this image for ubuntu22
 #  DOCKER_BUILDKIT=1 docker build --rm=true -f docker/dockerfiles/toolchains/toolchains-ubuntu.Dockerfile -t toolchains-ubuntu:latest docker/dockerfiles/toolchains/
 
+ARG GCC_INSTALL_HOME=/opt/gcc-toolset-14
+ARG GCC_WORK_DIR=/workspace/gcc-14
+ARG GCC_DOWNLOAD_URL=https://ftp.gnu.org/gnu/gcc/gcc-14.3.0/gcc-14.3.0.tar.gz
```

Comment:
centos7 use gcc-14.2, can these two platform uses the same gcc version?

Reply:
or consider move to ubuntu 24.04?

Reply:
Compared to gcc-14.2, version 14.3 has fixed several bugs. Therefore, I have chosen to upgrade to the newer version.

Reply:
I mean upgrade the build env to ubuntu 24.04 since v4.1, ubuntu 22.04 will be end of life on 2027, it's close.

Reply:
> Compared to gcc-14.2, version 14.3 has fixed several bugs. Therefore, I have chosen to upgrade to the newer version.


This primarily concerns the fix. https://gcc.gnu.org/bugzilla/show_bug.cgi?format=multiple&id=114945  This bug causes us to generate a significant number of warnings.